terraform/services/api-rds/main.tf

```
@@ -2,47 +2,132 @@ locals {
   db_name = {
     ab2d = {
       dev  = "ab2d-dev"
-      test = "ab2d-east-prod-test"
-      sbx  = "ab2d-sbx-sandbox"
+      test = "ab2d-east-impl"
       prod = "ab2d-east-prod"
-    }
-    bcda = {
-      dev  = "${var.app}-${var.env}"
-      test = "${var.app}-${var.env}"
-      sbx  = "${var.app}-${var.env}"
-      prod = "${var.app}-${var.env}"
-    }
-    dpc = {
-      dev  = "${var.app}-${var.env}"
-      test = "${var.app}-${var.env}"
-      sbx  = "${var.app}-${var.env}"
-      prod = "${var.app}-${var.env}"
-    }
+      sbx  = "ab2d-sbx-sandbox"
+    }[var.env]
+    bcda = "${var.app}-${var.env}"
+    dpc  = "${var.app}-${var.env}"
+  }[var.app]
+}
+
+## Begin module/main.tf
+
+# Create database security group
+resource "aws_security_group" "sg_database" {
+  name        = "${local.db_name}-database-sg"
+  description = "${local.db_name} database security group"
+  vpc_id      = data.aws_vpc.target_vpc.id
+  tags = merge(
+    data.aws_default_tags.data_tags.tags,
+    tomap({ "Name" = "${local.db_name}-database-sg" })
+  )
+
+  lifecycle {
+    create_before_destroy = true
   }
 }
 
+resource "aws_vpc_security_group_egress_rule" "egress_all" {
+  security_group_id = aws_security_group.sg_database.id
+
+  description = "Allow all egress"
+  cidr_ipv4   = "0.0.0.0/0"
+  ip_protocol = -1
+}
+
+resource "aws_vpc_security_group_ingress_rule" "db_access_from_jenkins_agent" {
+  description                  = "Jenkins Agent Access"
+  from_port                    = "5432"
+  to_port                      = "5432"
+  ip_protocol                  = "tcp"
+  referenced_security_group_id = var.jenkins_security_group_id
+  security_group_id            = aws_security_group.sg_database.id
+}
+
+resource "aws_vpc_security_group_ingress_rule" "db_access_from_controller" {
+  description                  = "Controller Access"
+  from_port                    = "5432"
+  to_port                      = "5432"
+  ip_protocol                  = "tcp"
+  referenced_security_group_id = data.aws_security_group.controller_security_group_id.id
+  security_group_id            = aws_security_group.sg_database.id
+}
+
+# Create database subnet group
+
+resource "aws_db_subnet_group" "subnet_group" {
+  name       = "${local.db_name}-rds-subnet-group"
+  subnet_ids = [data.aws_subnet.private_subnet_a.id, data.aws_subnet.private_subnet_b.id]
+}
+
+# Create database parameter group
+
+resource "aws_db_parameter_group" "parameter_group" {
+  name   = "${local.db_name}-rds-parameter-group-v15"
+  family = "postgres15"
+
+  parameter {
+    name         = "backslash_quote"
+    value        = "safe_encoding"
+    apply_method = "immediate"
+  }
+  parameter {
+    name         = "shared_preload_libraries"
+    value        = "pg_stat_statements,pg_cron"
+    apply_method = "pending-reboot"
+  }
+  parameter {
+    name         = "cron.database_name"
+    value        = local.db_name
```

I just realized that this does need to be a short name for ab2d databases, which is why they have [this map](https://github.com/CMSgov/ab2d-ops/blob/396c802e95381555230ee64043ecdeac597d968b/terraform/modules/data/main.tf#L2C3-L7) in ab2d-ops. Luckily, the short name is pretty close to our envs, so I think we can manage it like so:

```suggestion
    value        = var.app == "ab2d" && var.env == "test" ? "impl" : var.env
```

---

That makes sense! We can talk about why this is when I get back but for now I think this handles it fine to be functional. :)